src/os/keyvaluestore/KeyValueStore.cc

```diff
@@ -1213,9 +1213,8 @@ int KeyValueStore::_do_transactions(list<Transaction*> &tls, uint64_t op_seq,
   for (list<Transaction*>::iterator p = tls.begin();
        p != tls.end();
        ++p, trans_num++) {
-    r = _do_transaction(**p, bt, handle);
-    if (r < 0)
-      break;
+    unsigned ret = _do_transaction(**p, bt, handle);
+    assert(ret == 0);
```

Comment:
@dzafman Dear david, I am glad that you are back to work:-).  I revert this commit and just change the returned type of _do_transaction instead to better suit our expectation. Please help me to have a look at it again.